### PR TITLE
Added extraTokenParams to UserManager.signinSilent

### PR DESCRIPTION
**Pull request by MathiasRossen**
Closes/fixes [#1257](https://github.com/authts/oidc-client-ts/issues/1257)

### Checklist

- [x] This PR makes changes to the public API 
- [x] I have included links for closing relevant issue numbers

### CODE REVIEW

File: src/UserManager.ts [hunk 1]
```diff
@@ -283,14 +283,15 @@ export class UserManager {
         const {
             silentRequestTimeoutInSeconds,
             resource,
+            extraTokenParams,
```
Comment (pamapa) [inline src/UserManager.ts:286]: if you take it out like this, it will be missing for the iframe usecase.

File: src/UserManager.ts [hunk 2]
```diff
@@ -324,10 +325,11 @@ export class UserManager {
         return user;
     }
 
-    protected async _useRefreshToken(state: RefreshState): Promise<User> {
+    protected async _useRefreshToken(state: RefreshState, extraTokenParams?: Record<string, unknown>): Promise<User> {
```
Comment (pamapa) [inline src/UserManager.ts:328]: `protected async _useRefreshToken(state: RefreshState): Promise<User> {`
->
```ts
protected async _useRefreshToken(args: UseRefreshTokenArgs): Promise<User> {
  const response = await this._client.useRefreshToken({
    ...args,
    timeoutInSeconds: this.settings.silentRequestTimeoutInSeconds,
  });
```

**Comment by MathiasRossen** [comment]
Thank you, I will make some changes to resolve the issues

**Review by pamapa — APPROVED** [review]

**Comment by pamapa** [comment]
thanks for contributing

**Comment by MathiasRossen** [comment]
Cheers! I am glad it was possible and thanks for your fast response 🥳

**Pull request merged**